### PR TITLE
TRISA Protocol Buffers

### PR DESCRIPTION
**Pull request by bbengfort**
**Step 1: Redefine the TRISA protocol**

The original TrisaPeer2Peer service only implemented a transaction stream where VASPs could send transactions at each other but didn't really facilitate other interactions such as signing key exchange or address confirmation. Additionally, the protocol did not define errors or error handling. Steps taken:

1. Rename the service from `TrisaPeer2Peer` to `TRISANetwork` 
2. Add errors to the transaction and standardized error codes

### CODE REVIEW

File: proto/trisa/protocol/v1alpha1/api.proto [hunk 1]
```diff
@@ -0,0 +1,194 @@
+syntax = "proto3";
```
Comment (bbengfort) [inline proto/trisa/protocol/v1alpha1/api.proto:1]: @rebeccabilbo I've flushed out the TRISA protocol in more detail based on my reading of the white paper; I'd appreciate it if you could take a look and give me some feedback.

File: proto/trisa/protocol/v1alpha1/api.proto [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    // To conduct an information exchange prior to a virtual asset transaction, an
+    // originating VASP will send an encrypted transaction envelope to the beneficiary
+    // VASP containing a unique ID for the transaction, the encrypted transaction bundle,
+    // and metadata associated with the transaction cipher. In response, the beneficiary
+    // will validate the transaction request, then return the beneficiary's transaction
+    // information using the same unique transaction ID. The TRISANetwork provides both
+    // a unary RPC for simple, single transactions and a transaction stream for high
+    // throughput transaction workloads.
+    rpc Transfer(Transaction) returns (Transaction) {}
+    rpc TransactionStream(stream Transaction) returns (stream Transaction) {}
```
Comment (rebeccabilbro) [inline proto/trisa/protocol/v1alpha1/api.proto:28]: Great feature to have both unary and bidirectional streams to suit a range of developer use cases!

File: proto/trisa/protocol/v1alpha1/errors.proto [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        // fields is specified in extra.
+        MISSING_FIELDS = 153;
+
+        // The identity record is not complete enough for compliance purposes of the
+        // receiving VASPs. Required fields or format specified in extra.
+        INCOMPLETE_IDENTITY = 154;
+
+        // There was an error validating a field in the transaction data (specified in extra)
+        VALIDATION_ERROR = 155;
+    }
```
Comment (rebeccabilbro) [inline proto/trisa/protocol/v1alpha1/errors.proto:116]: Appreciate you providing such a comprehensive list of errors, as well as matching them up, where applicable, to the [Sygna BVRCs](https://developers.sygna.io/reference#beneficiary-vasp-rejected-code)!

**Review by rebeccabilbro — APPROVED** [review]
Nice PR fully fleshing out the robot VASP implementation for TRISA.

**Pull request merged**